src/adapter.rs

```diff
@@ -855,6 +887,32 @@ impl<'a> Adapter<'a> for RustdocAdapter<'a> {
                     (ctx, neighbors)
                 }))
             }
+            "Function" | "Method" if matches!(edge_name.as_ref(), "parameters") => {
+                let _current_crate = self.current_crate;
+                let _previous_crate = self.previous_crate;
+                Box::new(data_contexts.map(move |ctx| {
+                    let neighbors: Box<dyn Iterator<Item = Self::DataToken> + 'a> =
+                        match &ctx.current_token {
+                            None => Box::new(std::iter::empty()),
+                            Some(token) => {
+                                let origin = token.origin;
+                                let decl = if current_type_name.as_ref() == "Function" {
+                                    &token.as_function().expect("token was not a Function").decl
+                                } else {
+                                    &token.as_method().expect("token was not a Method").decl
+                                };
```

Comment:
As another option:
```suggestion
                                let decl = token.as_function().map(|f| &f.decl).unwrap_or(&token.as_method().expect("token was neither a Function nor a Method").decl);
```

The key difference is that the current code assumes that `current_type_name.as_ref() != "Function"` implies `current_type_name.as_ref() == "Method"`. This is currently true, because there are no edges that point directly to `FunctionLike`, but it's also a bit of a trap and a bug waiting to happen as soon as that sort of edge is added to the schema.

Recall that `current_type_name` here is the statically-known type of the vertex based on the query which includes `FunctionLike` as an option, not the specific type of a specific vertex which must be either `Function` or `Method` and can't be `FunctionLike`.